Cabal/Distribution/PackageDescription/Configuration.hs

```diff
@@ -416,15 +418,15 @@ newtype TargetSet a = TargetSet [(DependencyMap, a)]
 
 -- | Combine the target-specific dependencies in a TargetSet to give the
 -- dependencies for the package as a whole.
-overallDependencies :: TargetSet PDTagged -> DependencyMap
-overallDependencies (TargetSet targets) = mconcat depss
+overallDependencies :: ComponentEnabledSpec -> TargetSet PDTagged -> DependencyMap
+overallDependencies enabled (TargetSet targets) = mconcat depss
   where
     (depss, _) = unzip $ filter (removeDisabledSections . snd) targets
     removeDisabledSections :: PDTagged -> Bool
-    removeDisabledSections (Lib _ l)     = buildable (libBuildInfo l)
-    removeDisabledSections (Exe _ e)   = buildable (buildInfo e)
-    removeDisabledSections (Test _ t)  = testEnabled t && buildable (testBuildInfo t)
-    removeDisabledSections (Bench _ b) = benchmarkEnabled b && buildable (benchmarkBuildInfo b)
+    removeDisabledSections (Lib _ l)   = componentEnabled enabled (CLib l)
+    removeDisabledSections (Exe _ e)   = componentEnabled enabled (CExe e)
```

Comment:
To recap:

- `componentEnabled` is whether I asked for component to be built
- `componentBuildable` is whether component is buildable in my environment (i.e. package author)

AFAICS there aren't currently a possiblity to *enable* component which isn't *buildable* (i.e. we say *enable whatever you can *build(), but in the future we might run into that situation?

If this were GHC, I'd add a `[Note]` and refer to it from both `componentEnabled` and `componentBuildable` docs.

*EDIT:* `componentBuildable` could depend only on the environmen (`os`, `impl` etc)?